packages/flutter_tools/lib/src/base/config.dart

```diff
@@ -92,9 +103,24 @@ class Config {
   //
   // Note that this is different from FileSystemUtils.homeDirPath.
   static String _userHomePath(Platform platform) {
-    final String envKey = platform.operatingSystem == 'windows'
-      ? 'APPDATA'
-      : 'HOME';
+    final String envKey =
+        platform.operatingSystem == 'windows' ? 'APPDATA' : 'HOME';
     return platform.environment[envKey] ?? '.';
   }
+
+  static String _configPath(
+      Platform platform, FileSystem fileSystem, String name) {
+    final String homeDirFile =
+        fileSystem.path.join(_userHomePath(platform), '.${kConfigDir}_$name');
+    if (!platform.isLinux && !platform.isMacOS) {
+      return homeDirFile;
+    }
+    if (fileSystem.isFileSync(homeDirFile)) {
+      return homeDirFile;
+    } else {
+      final String configDir = platform.environment['XDG_CONFIG_HOME'] ??
```

Comment:
pull out `XDG_CONFIG_HOME` into a constant and document it with a link to the specification